packages/starlight/components/MarkdownContent.astro

```diff
@@ -2,112 +2,139 @@
 
 <style>
   .content
-    :global(
+    > :global(
       :not(a, strong, em, del, span, input)
-        + :not(a, strong, em, del, span, input)
+        + :not(
+          a,
+          strong,
+          em,
+          del,
+          span,
+          input,
+          :where([class~='not-content'] *)
+        )
     ) {
     margin-top: 1.5rem;
   }
 
   /* Headings after non-headings have more spacing. */
-  .content :global(:not(h1, h2, h3, h4, h5, h6) + :is(h1, h2, h3, h4, h5, h6)) {
+  .content
+    :global(
+      :not(h1, h2, h3, h4, h5, h6)
+        + :is(h1, h2, h3, h4, h5, h6):not(:where([class~='not-content'] *))
+    ) {
     margin-top: 2.5rem;
   }
 
-  .content :global(li + li),
-  .content :global(dt + dt),
-  .content :global(dt + dd),
-  .content :global(dd + dd) {
+  .content :global(li + li:not(:where([class~='not-content'] *))),
+  .content :global(dt + dt:not(:where([class~='not-content'] *))),
+  .content :global(dt + dd:not(:where([class~='not-content'] *))),
+  .content :global(dd + dd:not(:where([class~='not-content'] *))) {
     margin-top: 0.25rem;
   }
 
   .content
     :global(
-      li > :last-child:not(li, ul, ol):not(a, strong, em, del, span, input)
+      li
+        > :last-child:not(li, ul, ol):not(
+          a,
+          strong,
+          em,
+          del,
+          span,
+          input,
+          :where([class~='not-content'] *)
+        )
     ) {
     margin-bottom: 1.25rem;
   }
 
-  .content :global(dt) {
+  .content :global(dt:not(:where([class~='not-content'] *))) {
     font-weight: 700;
   }
-  .content :global(dd) {
+  .content :global(dd:not(:where([class~='not-content'] *))) {
     padding-inline-start: 1rem;
   }
 
-  .content :global(:is(h1, h2, h3, h4, h5, h6)) {
+  .content
+    :global(:is(h1, h2, h3, h4, h5, h6):not(:where([class~='not-content'] *))) {
     color: var(--sl-color-white);
     line-height: var(--sl-line-height-headings);
     font-weight: 600;
   }
 
-  .content :global(:is(img, picture, video, canvas, svg, iframe)) {
+  .content
+    :global(
+      :is(img, picture, video, canvas, svg, iframe):not(
+          :where([class~='not-content'] *)
+        )
+    ) {
     display: block;
     max-width: 100%;
     height: auto;
   }
 
-  .content :global(h1) {
+  .content :global(h1:not(:where([class~='not-content'] *))) {
```

Comment:
The `*` looks like it might throw this selector off but I can't quite place my finger on it 😅 

Is this making sure the `h1` style won't be applied if any parent has the `not-content` class?

I think that may need to be done by *undoing* the style when a `.not-content` parent is found since the `h1` selector itself couldn't look back up the tree

Something like,

```css
.content :global(h1) {
  font-size: var(--sl-text-h1);
}

.content :global(.not-content h1) {
  font-size: unset; /* or initial? */
}
```

Reply:
Yeah this combo is not super easy to read but to pull it apart:

```css
h1:not(:where([class~='not-content'] *)) {}
```

Select all `h1` elements which are not _also_ descendents of an element with the `not-content` class. `:where()` effectively let’s you do an intersection of conditions, which is pretty powerful when combined with `:not()`!

Aside: realising now that the attribute selector (`[class~=...]`) has the same specificity as a plain `.not-content`, so might as well just use that I think.

```css
h1:not(:where(.not-content *)) {}
```

Reply:
Well played, that's some magic right there!  I never realized you could effectively do a check up the tree with a combination of `:not()` and `:where()`!  🪄 👏